lib/AgrirouterClient/src/Protobuf/meson.build

```diff
@@ -0,0 +1,52 @@
+protoc_exe = find_program('protoc')
+
+pymodule = import('python3')
+python3 = pymodule.find_python()
+
+res = run_command(python3, '-c', 'import os; print(os.environ["PKG_CONFIG_SYSROOT_DIR"])')
+if res.returncode() == 0
+  sdk_var = res.stdout().strip() + '/usr/include/'
+else 
+  sdk_var = meson.current_source_dir()
+endif
+
+protobuf_generator = generator(
+    protoc_exe,
+    output    : ['@BASENAME@.pb.cc', '@BASENAME@.pb.h'],
+    arguments : [
+      '-I=' + meson.current_source_dir(),
+      '-I=' + sdk_var,
+      '--cpp_out=@BUILD_DIR@',
+      '@INPUT@'
+      ]
+)
+
+agrirouter_protobuf_files = protobuf_generator.process(
+    join_paths([meson.current_source_dir(), 'commons/message.proto']),
```

Comment:
Any particular reason you need meson.current_source_dir here?

Reply:
When I delete it, the message appears:
"FAILED: lib/AgrirouterClient/src/Protobuf/libagrirouter_proto.a.p/commons/message.pb.cc lib/AgrirouterClient/src/Protobuf/libagrirouter_proto.a.p/commons/message.pb.h 
/usr/local/bin/protoc -I=/home/xx/yyy/agrirouter-api-cpp/lib/AgrirouterClient/src/Protobuf -I=/home/xx/yyy/agrirouter-api-cpp/lib/AgrirouterClient/src/Protobuf --cpp_out=lib/AgrirouterClient/src/Protobuf/libagrirouter_proto.a.p ../lib/AgrirouterClient/src/Protobuf/commons/message.proto
../lib/AgrirouterClient/src/Protobuf/commons/message.proto: File does not reside within any path specified using --proto_path (or -I).  You must specify a --proto_path which encompasses this file.  Note that the proto_path must be an exact prefix of the .proto file names -- protoc is too dumb to figure out when two paths (e.g. absolute and relative) are equivalent (it's harder than you think)."

I've try to set a --proto_path to meson.current_source_dir and then delete in the file list, but it is the message.
Do you please have a hint for me how to solved it in the correct way?

Reply:
>  Note that the proto_path must be an exact prefix of the .proto file names -- protoc is too dumb to figure out when two paths (e.g. absolute and relative) are equivalent (it's harder than you think)."

Oh hmm, disappointing.

I guess that what you want here is the --proto_path equaling the relativity of the build root to the source root.

There's a proposal to add a relative paths method to the `fs = import('fs')` module, but that's not merged yet. However you can do this:

```meson
build_to_source = run_command(
    'python3', '-c',
    'import sys, os; print(os.path.relpath(sys.argv[1], sys.argv[2]))',
    meson.project_source_root(), meson.project_build_root(),
    check: true).stdout().strip()
```

Then `build_to_source` will be the correct path to strip off of these files. It will probably be something like `../` if the build directory is a subdirectory of the source directory -- either way, it will be the same "path to meson.current_source_dir()" that is being written into the build commands as `./lib/AgrirouterClient/src/Protobuf/commons/message.proto`.

Reply:
I've been trying for a few hours now. but I can't get it to work any better than it is now. It always comes the message as in the first comment of mine.

I think I don't understand how to put the build_to_source path.

If I take out the meson.current_source_dir() then "/home/xx/yy/agrirouter-api-cpp/lib/AgrirouterClient/src/Protobuf/commons/message.proto" becomes "../lib/AgrirouterClient/src/Protobuf/commons/message.proto".

I need a second hint to figure out how i can compile the protobuf without the join paths